src/components/useraccount/changepassword/TabChangePassword.vue

```diff
@@ -0,0 +1,154 @@
+              </div>
+            </div>
+          </div>
+          <div class="row">
+            <div class="col-sm-6">
+              <div class="form-sections">
+                <BaseInput v-model="form.newPassword"
+                           :vuelidate="$v.form.newPassword"
+                           :label="$t('newPassword')"
+                           type="password"
+                           class="form-inputs"
+                           data-test="change-password-form-newpassword"/>
+              </div>
+            </div>
+            <div class="col-sm-6">
+              <div class="form-sections">
+                <BaseInput v-model="form.newPasswordConfirm"
+                           :vuelidate="$v.form.newPasswordConfirm"
+                           :label="$t('newPasswordConfirm')"
+                           :customErrors="{ sameAsPassword: $t('newPasswordConfirmError') }"
+                           type="password"
+                           class="form-inputs"
+                           data-test="change-password-form-newpasswordconfirm"/>
+              </div>
+            </div>
+          </div>
+          <hr class="light-grey-hr">
+          <div class="personal-details-edit-btn">
+            <span>
+              <LoadingButton :state="state"
+                             :disabled="!$v.$anyDirty"
+                             type="submit"
+                             class="update-btn"
+                             data-test="edit-profile-form-submit">
+                {{ $t('updateBtn') }}
+              </LoadingButton>
+            </span>
+          </div>
+        </BaseForm>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import { required, sameAs } from 'vuelidate/lib/validators';
+import gql from 'graphql-tag';
+import customerMixin from '../../../mixins/customerMixin';
+import ServerError from '../../common/form/ServerError.vue';
+import LoadingButton from '../../common/form/LoadingButton.vue';
+import BaseInput from '../../common/form/BaseInput.vue';
+import BaseForm from '../../common/form/BaseForm.vue';
+
+export default {
+
+  components: {
+    BaseForm,
+    BaseInput,
+    LoadingButton,
+    ServerError,
+  },
+
+  mixins: [customerMixin],
+
+  data: () => ({
+    me: null,
+    form: {},
+  }),
+
+  methods: {
+    updateCustomerPassword() {
+      return this.updateMyCustomerPassword(this.form.currentPassword, this.form.newPassword)
+        .then(() => {
+          this.form = {};
+          this.$v.$reset();
+        });
+    },
+
+    getErrorMessage({ code }) {
+      if (code === 'InvalidCurrentPassword') {
+        return this.$t('invalidPassword');
+      }
+      return this.$t('unknownError');
+    },
+  },
+
+  apollo: {
+    me: {
+      query: gql`
+        query me {
+          me {
+            customer {
+              id
+              version
+            }
+          }
+        }`,
+      skip: vm => !vm.$store.state.authenticated,
+    },
+  },
```

Comment:
Are you sure you need this apollo request? If it's only this, the one in `customerMixin` should do fine.

Reply:
I actually didn't need but now I do, as I have login in the component and for that I need email, you can check it in the new PR

Reply:
You could have still used the approach to get the email from the response, but it's true that it looks a bit dirty to navigate through the response. But it's something that should be enabled (so that you can react to errors). Anyway, it's good also this way.